src/inference/gibbs.jl

```diff
@@ -30,11 +39,16 @@ Tips:
 methods like Particle Gibbs. You can increase the effectiveness of particle sampling by including
 more particles in the particle sampler.
 """
-struct Gibbs{space, A<:Tuple{Vararg{GibbsComponent}}} <: InferenceAlgorithm
+struct Gibbs{space, A<:Tuple} <: InferenceAlgorithm
     algs::A   # component sampling algorithms
+
+    function Gibbs{space, A}(algs::A) where {space, A<:Tuple}
+        @assert all(isgibbscomponent, algs)
```

Comment:
`@assert` can be removed at various optimization levels (see https://docs.julialang.org/en/v1/base/base/#Base.@assert), so it is safer to use something like
```suggestion
        all(isgibbscomponent, algs) || error("all algorithms have to support Gibbs sampling")
```